File: WorkoutTracker.Domain/Entities/Workouts/Workout.cs
using WorkoutTracker.Domain.Base;
using WorkoutTracker.Domain.Entities.Exercises.ExerciseSets;

namespace WorkoutTracker.Domain.Entities.Workouts;

public class Workout : BaseWorkoutModel
{
    public string? Description { get; set; }
    public DateTime Created { get; set; }
    public bool IsPinned { get; set; }
    public int CountOfTrainings { get; set; }

    public string UserId { get; set; } = null!;
    public IEnumerable<ExerciseSetGroup>? ExerciseSetGroups { get; set; }
    public IEnumerable<WorkoutRecord>? WorkoutRecords { get; set; }
}

File: WorkoutTracker/src/app/workouts/models/workout.ts
import { Exercise } from "src/app/exercises/models/exercise";
import { ExerciseSetGroup } from "src/app/shared/models/exercises/exercise-set-group";
import { ModelWeight } from "src/app/shared/models/model-weight";
import { WorkoutModel } from "src/app/shared/models/base/workout-model";

export interface Workout extends WorkoutModel {
    description: string|null;
    created: Date;
    started: Date|null;
    isPinned: boolean;
    weight: ModelWeight;

    exercises: Exercise[];
    exerciseSetGroups: ExerciseSetGroup[];
}